### ISSUE TITLE
error handling

### ISSUE THREAD

**Issue by vitaly-t** [jmeas/finance-app#138]
for this code: https://github.com/jmeas/finance-app/blob/master/server/api/util/catch-rejected-query.js

It is probably better to handle such errors in a more generic way, because error messages get refactored sometimes, which would break your code.

```js
'use strict';

const generateErrors = require('../errors/generate-errors');
const QueryResultError = require('pg-promise').errors.QueryResultError;

// Our generic `.catch()` for failed promises from pg-query
function catchRejectedQuery(res, e) {
    if (e instanceof QueryResultError) {
        res.status(404).send({
            // Unexpected amount of data returned (against the query expectation)
            
            // errors: [generateErrors.notFoundError()]
        });
    } else {
        res.status(500).send({
            errors: [generateErrors.genericError()]
        });
    }
}

module.exports = catchRejectedQuery;
```

**Comment by jmeas** [comment]
Excellent suggestion, as usual, @vitaly-t . I've taken a note; this will be refactored soon!

**Comment by vitaly-t** [comment]
I'm going to make another release later today, one that will bring error codes for type `QueryResultError`, in case someone wants to handle those errors on a more granular level.

**Comment by jmeas** [comment]
yessss that would be amazing!

**Comment by vitaly-t** [comment]
@jmeas Done & dusted, see [release 3.6.0](https://github.com/vitaly-t/pg-promise/releases/tag/v.3.6.0).

**Comment by vitaly-t** [comment]
@jmeas now with [3.7.0](https://github.com/vitaly-t/pg-promise/releases/tag/v.3.7.0) out, we all should be happy :)

**Comment by jmeas** [comment]
Nice. I plan to upgrade this weekend.

**Comment by jmeas** [comment]
Resolved in #147
